Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/DocumentAdapter.h

```diff
@@ -60,10 +62,37 @@ namespace AZ::DocumentPropertyEditor
         //! \see RoutingAdapter
         virtual void SetRouter(RoutingAdapter* router, const Dom::Path& route);
 
+        //! If true, debug mode is enabled for all DocumentAdapters.
+        //! \see SetDebugModeEnabled
+        static bool IsDebugModeEnabled();
+        //! Enables or disables debug mode globally for all DocumentAdapters.
+        //! Debug mode adds expensive extra steps to notification operations to ensure the
+        //! adapter is behaving correctly and will report warnings if an issue is detected.
+        //! This can also be set at runtime with the `ed_debugDocumentPropertyEditorUpdates` CVar.
+        static void SetDebugModeEnabled(bool enableDebugMode);
+
     protected:
+        //! Generates the contents of this adapter. This must be an Adapter DOM node.
+        //! These contents will be cached - to notify clients of changes to the structure,
+        //! NotifyResetDocument or NotifyContentsChanged must be used.
+        //! \see AdapterBuilder for building out this DOM structure.
+        virtual Dom::Value GenerateContents() const = 0;
+
+        //! Specifies the type of reset operation triggered in NotifyResetDocument.
+        enum class DocumentResetType
+        {
+            //! (Default) On soft reset, the adapter will compare any existing cached contents to the new result of
+            //! GetContents and produce patches based on the difference, assuming cached contents are available.
+            SoftReset,
```

Comment:
I feel that the concept of SoftReset makes this API hard to understand. To keep things simple, I recommend keeping only one reset that acts like the HardReset. For the cases where you need the functionality of SoftReset, use a different function like 'CompareStatesAndGeneratePatches' or something that clearly indicates what is actually going on.

Reply:
We chatted about this a bit offline, but for reference the answer is "mostly, don't worry about it" - if you want to reset your adapter, you generally don't need to worry about the implementation detail of how the adapter goes about notifying views of its new contents, unless you're performance profiling and hit a hot spot, then this documentation can come in handy. These aren't *logically* different methods, they both do the same thing, they just take different paths to accomplish it.